### PR TITLE
implement kill tests

### PR DESCRIPTION
**Pull request by liangchenye**
Signed-off-by: Liang Chenye <liangchenye@huawei.com>

### CODE REVIEW

**Comment by liangchenye** [comment]
- add 'kill' interface to container.go
  our interface docucment [--signal <SIGNAL>](https://github.com/opencontainers/runtime-tools/blob/master/docs/command-line-interface.md#kill) is different with current runc.
   I'm using runc way now.
- fix bug in test.go
- add kill.go 
  I did not find a good way to test this spec error ` Attempting to send a signal to a container that is neither created nor running MUST have no effect on the container`, so it is missing in this PR.
- add killsig.go
   now  'TERM' and 'KILL' are MUST supported, 'KILL' is verified in 'kill.go'.
   In killsig.go, we only verify 'TERM' now.
   In the future we might add more signal tests.  

Tested in my environment with 'runc' , it works.
PTAL  @wking @q384566678

File: validation/kill.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+				r.SetID(c.id)
+				return nil
+			},
+			PreDelete: func(r *util.Runtime) error {
+				// waiting the 'stoppedConfig' to stop
+				// the running test case sleeps 30 seconds, so 10 seconds are enough for this case
+				util.WaitingForStatus(*r, util.LifecycleStatusCreated|util.LifecycleStatusStopped, time.Second*10, time.Second*1)
+				// KILL MUST be supported and KILL cannot be trapped
+				err := r.Kill("KILL")
+				// wait before the container been deleted
```
Comment (wking) [inline validation/kill.go]: I dont think the next line needs a comment; `util.WaitingForStatus(*r, util.LifecycleStatusStopped, ...)` is pretty clear on it's own.  And if you do want a comment, you probably want "been deleted" ->"has stopped".

Comment (liangchenye) [reply validation/kill.go]: I remove this line, updated.

**Comment by liangchenye** [comment]
Another issue I found, 'util.Fatal' exits immediately, the 'defer' functions will not be called.
So the bunde dir might not be removed and the container might not be deleted.
It needs to be fixed.

File: validation/killsig.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+var signals = []string{
+	"TERM",
+}
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+	bundleDir, err := util.PrepareBundle()
+	fmt.Println(bundleDir)
```
Comment (q384566678) [inline validation/killsig.go]: Should delete it?

Comment (liangchenye) [reply validation/killsig.go]: updated

**Comment by q384566678** [comment]
LGTM

**Pull request merged**